src/app/components/search-field/search-field.component.html

```diff
@@ -0,0 +1,13 @@
+<onto-search-field
+  [autocompleteData]="states"
+  [customTemplate]="useCustomTemplate() ? customTemplate : null"
+  [selectedList]="getSelectedList()"
```

Comment:
When you pass a function to an Input it executes on every change detection.
What is the benefit here of passing functions? Do you change the values of the inputs dynamicaly?